resources/scripts.js

```diff
@@ -125,112 +188,362 @@ function buttonPosition(loc) {
-    displayMessage('You lose')
+function payOut(isprimary, win, amount) {
+    if (isprimary === false) {
+        if (win === true) {
+            playerMoney += (amount * 2);
+            totalBets -= amount;
+            return;
+        }
+        totalBets -= amount;
+        return;
+    }
+    if (win === true) {
+        buildSummary(PRIMARYWIN);
+        lastPlayerMoney = playerMoney;
+        playerMoney += (amount * 2);
+        pushSecondaryBets();
+        resetGame();
+        moneyChange(0);
+        return;
+    }
+    buildSummary(PRIMARYLOSE);
     if (playerMoney === 0) {
         gameOver();
         return;
     }
-    buttonPosition(1);
-    betAmount = 0;
-    rollButtonState(false);
-    betButtonState(true);
+    displayMessage(rollSummary);
+    rollSummary = null;
+    buttonPosition(playerRoll);
+    resetGame();
+    moneyChange(0);
+    return;
 }
 
-function placeBet() {
-    betAmount = betDialog.value;
+function makeBet() {
+    betAmount = betDialogElement.value;
     betAmount = parseInt(betAmount);
-    if (validateBet() === false) {
+    if (validateBet(betAmount) === false) {
         return;
     }
+    lastPlayerMoney = playerMoney;
     playerMoney -= betAmount;
-    displayMoney.innerHTML = playerMoney;
-    rollButtonState(true);
-    betButtonState(false);
+    moneyChange(betAmount);
+    buttonStates();
     return;
 }
 
+function makeCome() {
+    let comeAmount = betDialogElement.value;
+    comeAmount = parseInt(comeAmount);
+    if (validateBet(comeAmount) === false) {
+        return;
+    }
+    lastPlayerMoney = playerMoney;
+    playerMoney -= comeAmount;
+    secondaryBetObj[1] = comeAmount;
+    moneyChange(comeAmount);
+    betDialogElement.disabled = true;
+    comeButtonElement.disabled = true;
+    return;
+}
+
+function makePlace(num) {
+    placeDialogAmount.setAttribute('max', playerMoney);
+    makePlaceNumClicked = num;
+    placeDialog.showModal();
+    return;
+}
+
+class SecondaryBet {
+    constructor(point, amount) {
+        this.point = point;
+        this.amount = amount;
+    }
+}
+
+function pushSecondaryBets() {
+    if (testObjPop(secondaryBetObj) === false) {
+        for (let i = 1; i < 11; i++) {
+            if (secondaryBetObj[i] > 0) {
+                playerMoney += secondaryBetObj[i];
+                totalBets -= secondaryBetObj[i];
+                delete secondaryBetObj[i];
+            }
+        }
+    }
+    return;
+}
+
+
 function gameOver() {
-    displayMessage(`You are bankrupt! Please choose 'New Game' from the menu to play again`);
-    buttonPosition(1);
-    rollButtonState(false);
-    betButtonState(false);
+    displayMessage(BANKRUPT);
+    playerRoll = 1;
+    buttonPosition(playerRoll);
+    buttonStates();
+    return;
+}
+
+async function moneyChange(newBet) {
+    totalBets += newBet;
+    playerMoneyElement.innerHTML = playerMoney;
+    playerBetElement.innerHTML = totalBets;
+    if (playerMoney === lastPlayerMoney) {
```

Comment:
I'm not sure this if statement is needed because the next two if statements cover the other 2 possibilities and nothing happens anyway when the money is equal

Reply:
Removed code and tested, appears to work as expected after removal